src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java

```diff
@@ -501,148 +1387,699 @@ private boolean isAmqpHeaderValid(
                 header.revision() == 0;
         }
 
-        private void onAmqpOpen(
-            AmqpFrameFW open)
+        private void cleanupNetwork(
+            long traceId,
+            long authorization)
         {
-            // TODO
-            doAmqpOpen(open);
+            doNetworkReset(traceId, authorization);
+            doNetworkAbort(traceId, authorization);
+            cleanupStreams(traceId, authorization);
         }
 
-        private void onAmqpBegin(
-            AmqpFrameFW begin)
+        private void cleanupStreams(
+            long traceId,
+            long authorization)
         {
             // TODO
-            /*
-            * get a session from map with channel
-            * */
         }
 
-        private void onAmqpAttach(
-            AmqpFrameFW attach)
+        private void cleanupBudgetCreditorIfNecessary()
         {
-            // TODO
+            if (replyBudgetIndex != NO_CREDITOR_INDEX)
+            {
+                creditor.release(replyBudgetIndex);
+                replyBudgetIndex = NO_CREDITOR_INDEX;
+            }
         }
 
-        private void onAmqpFlow(
-            AmqpFrameFW flow)
+        private void cleanupDecodeSlotIfNecessary()
         {
-            // TODO
+            if (decodeSlot != NO_SLOT)
+            {
+                bufferPool.release(decodeSlot);
+                decodeSlot = NO_SLOT;
+                decodeSlotLimit = 0;
+            }
         }
 
-        private void onAmqpTransfer(
-            AmqpFrameFW transfer)
+        private void cleanupEncodeSlotIfNecessary()
         {
-            // TODO
+            if (encodeSlot != NO_SLOT)
+            {
+                bufferPool.release(encodeSlot);
+                encodeSlot = NO_SLOT;
+                encodeSlotOffset = 0;
+                encodeSlotTraceId = 0;
+            }
         }
 
-        private void onAmqpDisposition(
-            AmqpFrameFW disposition)
+        private final class AmqpSession
         {
-            // TODO
+            private AmqpSession()
+            {
+                // TODO
+            }
+
+            private void onDecodeBegin(
+                long traceId,
+                long authorization)
```

Comment:
We need the `AmqpBeginFW` as a parameter, then assert the `channelId` is as expected from constructor.

Reply:
According to the `AMQP` spec, _If a session is locally initiated, the remote-channel MUST NOT be set_. So when we receive `AmqpBeginFW` `begin` here, the `channelId` is null.

Reply:
Ah yes, thanks for the reminder. :-)